### ISSUE TITLE
io_uring / IoRing loop

### ISSUE THREAD

**Issue by staabm** [revoltphp/event-loop#81]
Would it make sense to implement a new based on io_uring / IoRing?

Found a discussion which shares other languages/applications to use a io_uring based loop for better performance, see https://news.ycombinator.com/item?id=23133040

IoRing on Windows: https://github.com/oven-sh/bun/issues/43#issuecomment-1605693525

[the origin idea](https://github.com/php/php-src/issues/8013#issuecomment-1357377716) for opening this issue is by Arnauld, a php-src developer

**Comment by WyriHaximus** [comment]
In short, if there is a way to access it from PHP through core functions or extensions, then yes it would make sense if we can get all supported features we expect from event loops in.

**Comment by kelunik** [comment]
This would be nice to have, but not at the top of my TODO list. Also keep in mind that uring is still quite young and people keep finding new vulnerabilities, so [Google disabled it on their production servers](https://security.googleblog.com/2023/06/learnings-from-kctf-vrps-42-linux.html?m=1).

**Comment by bartvanhoutte** [comment]
For what it's worth: (limited) support for io_uring was added in libuv a while ago: https://github.com/libuv/libuv/issues/1947.

**Comment by danog** [comment]
Yep, the uv backend already automatically uses io_uring when possible (causing permanent kernel memory leaks on specific Linux versions unless disabled via UV_USE_IO_URING :)))
